api/src/routes/edit.ts

```diff
@@ -126,6 +126,46 @@ function uploadFile(req, res, next) {
     });
 }
 edit.post("/object/:pid/datastream/:stream", requireToken, datastreamSanitizer, uploadFile);
+edit.post(
+    "/object/:pid/datastream/:stream/license",
+    requireToken,
+    bodyParser.json(),
+    datastreamSanitizer,
+    async function (req, res) {
+        const { pid, stream } = req.params;
+        const { licenseKey } = req.body;
+        try {
+            const datastream = DatastreamManager.getInstance();
+            await datastream.uploadLicense(pid, stream, licenseKey);
+            res.status(200).send("Upload license success");
+        } catch (error) {
+            console.log("error", error);
+            res.status(500).send(error.message);
+        }
+    }
+);
+
+edit.get("/object/:pid/datastream/:stream/license", requireToken, datastreamSanitizer, async (req, res) => {
+    try {
+        const { pid, stream } = req.params;
+        const datastream = DatastreamManager.getInstance();
+        const licenseKey = await datastream.getLicenseKey(pid, stream);
+        res.status(200).send(licenseKey);
+    } catch (error) {
+        res.status(500).send(error.message);
+    }
+});
+
+edit.get("/object/:pid/datastream/:stream/metadata", requireToken, datastreamSanitizer, async (req, res) => {
```

Comment:
Is this metadata route used for anything yet? I see we already defined `getObjectDatastreamMetadataUrl` in a previous pull request but haven't used it yet... and this seems related to that, but also not used yet. Should we move this to a separate PR for now, or is it part of the current work and I'm just missing something?

Reply:
I just added since it was useful for testing. Even though it's not in it's final form, I figured I might as well add it since getting the metadata is going to be part of future work anyway. Let me know if it produces a potential problem.

Reply:
As long as it's intentional, I don't object -- just wanted to be sure it wasn't an artifact that got added by accident. :-)